packages/nightly/src/lib/nightly.ts

```diff
@@ -0,0 +1,170 @@
+import type {
+  InjectedWallet,
+  InjectedWalletMetadata,
+  WalletBehaviourFactory,
+  WalletModuleFactory,
+} from "@near-wallet-selector/core";
+import { createAction } from "@near-wallet-selector/wallet-utils";
+import { isMobile } from "is-mobile";
+import { utils } from "near-api-js";
+import { JsonRpcProvider } from "near-api-js/lib/providers";
+import { createTransaction } from "near-api-js/lib/transaction";
+import { Account, waitFor } from "@near-wallet-selector/core";
+import type { NearNightly, NightlyInjected } from "./injected-nightly";
+import { PublicKey } from "near-api-js/lib/utils";
+
+declare global {
+  interface Window {
+    nightly: NightlyInjected | undefined;
+  }
+}
+
+interface NightlyState {
+  wallet: NearNightly;
+}
+
+const setupNightlyState = (meta: InjectedWalletMetadata): NightlyState => {
+  const wallet = window.nightly?.near;
+  if (!wallet) {
+    window.location.href = meta.downloadUrl;
```

Comment:
Is there a way to download this wallet extension so we can try it out?

Reply:
YES I left a link in readme. Please dont share for now 😈

Reply:
To align with other injected wallets, we need to avoid automatically redirecting like this. Wallet Selector will only display wallets that are currently installed. `downloadUrl` will likely be used in the future to handle this use case.

Take a look at the `isInstalled` logic for Sender and Math Wallet to return `null`

Reply:
Refactored so extension is still visible but will redirect only on click.

Reply:
This isn't a concern that should be handled by the wallet implementations but instead the UI such as `modal-ui`. As mentioned in my previous comment, the wallet should return `null` if it's not installed. The benefit here is we can be sure the wallet exists without introducing lots of `?.` checks.

They'll be some work in the future to address this particular issue in `modal-ui` where the `metadata.downloadUrl` will be used to redirect but this behaviour shouldn't be part of the wallet.